bluemail: init at 1.131.4-1795

onny opened the pull request:
###### Motivation for this change
Adds package and mail app [BlueMail](https://bluemail.me/), graphical desktop mail client which also includes a calendar.

Fixes https://github.com/NixOS/nixpkgs/issues/154270 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

onny committed:
bluemail: init at 1.131.4-1795

FliegendeWurst commented:
It seems you forgot to upload `pkgs/applications/networking/mailreaders/bluemail/default.nix` ?

onny commented:
> It seems you forgot to upload `pkgs/applications/networking/mailreaders/bluemail/default.nix` ?

Added the missing file :)

FliegendeWurst commented on pkgs/applications/networking/mailreaders/bluemail/default.nix:
```diff
@@ -0,0 +1,56 @@
+  version = "1.1.119";
+
+  src = fetchurl {
+    url = "https://download.bluemail.me/BlueMail/deb/BlueMail.deb";
```
This URL doesn't include the version. You may have to switch to an archived download if this is the only download link.

onny replied:
Not sure how to fix this. Should I mirror this file to Internet Archive for a persistent link?

Mikilio replied:
The snap version is usually more up-to-date and has a static link with a revision number.
You can get relevant information for the newest version with the following command:
`curl -H 'X-Ubuntu-Series: 16' https://api.snapcraft.io/api/v1/snaps/details/bluemail | jq '.version,.anon_download_url,.revision' -r`
The download link will usually look something like this:
`https://api.snapcraft.io/api/v1/snaps/download/ZVlj0qw0GOFd5JgTfL8kk2Y5eIG1IpiH_${revision}.snap`
This whole procedure has been done for the Spotify nix package as well, so maybe it can serve as a good reference.

I searched hard and couldn't find solutions for the deb and rpm packages.

onny replied:
Thanks @SuperSandro2000 that worked. I would like to prefer this approach since it is already working. I added a comment on how to manually update it in case there's a new version

FliegendeWurst commented:
Try this:

```
  nativeBuildInputs = [
    autoPatchelfHook
    dpkg
  ];

  buildInputs = [
    pango
    gtk3
    alsa-lib
    nss
    libXdamage
    libdrm
    mesa
    libxshmfence
  ];

  unpackCmd = "${dpkg}/bin/dpkg-deb -x $src debcontents";

  dontBuild = true;
  dontStrip = true;

  installPhase = ''
    mkdir -p $out
    mkdir $out/bin
    mv opt/BlueMail/* $out
    ln -s $out/bluemail $out/bin/bluemail
  '';
```

It builds fine, but I can't launch the app..

onny commented:
Thank you for figuring this out, maybe we'll can also fix the segfault somehow :)

Mikilio commented:
Not stale.

onny commented:
I switched to a snapshot of the release tarball so we'll have a fixed source. I prefer this approach since it is already working and easier to manage. I added a comment on how to update for later releases. Thank you all for reviewing and giving tips!

Mikilio reviewed (commented):
The fact that upgrading or downgrading (query all snapshots for the correct version) needs to be done by hand is not optimal, but I don't think it's a requirement for nixpkgs to provide those functions. If it is, maybe I could help with that by writing a script.
